### PR TITLE
SI-2707-vehicle-by-sd

### PR DESCRIPTION
**Pull request by Allyson-English**
find vehicle a synthetic device is connected to

### CODE REVIEW

**Review by KevinJoiner — APPROVED** [review]

File: graph/schema/synthetic.graphqls [hunk 1]
```diff
@@ -64,6 +64,10 @@ type SyntheticDevice implements Node {
   The block timestamp at which this device was minted.
   """
   mintedAt: Time!
+  """
+  The vehicle, if any, with which the synthetic device is paired.
+  """
+  vehicle: Vehicle
```
Comment (elffjs) [inline graph/schema/synthetic.graphqls:70]: Probably `Vehicle!` for now?

File: internal/repositories/synthetic/synthetic_devices.go [hunk 2]
```diff
@@ -91,25 +92,23 @@ func (r *Repository) GetSyntheticDevices(ctx context.Context, first *int, last *
 			return nil, fmt.Errorf("invalid after cursor: %w", err)
 		}
 		queryMods = append(queryMods, models.SyntheticDeviceWhere.ID.LT(afterID))
-	}
-
-	if before != nil {
+	} else if before != nil {
```
Comment (elffjs) [inline internal/repositories/synthetic/synthetic_devices.go:95]: Both can be specified. That's legal.

File: internal/repositories/synthetic/synthetic_devices.go [hunk 3]
```diff
@@ -91,25 +92,23 @@ func (r *Repository) GetSyntheticDevices(ctx context.Context, first *int, last *
...
 		beforeID, err := helpers.CursorToID(*before)
 		if err != nil {
 			return nil, fmt.Errorf("invalid before cursor: %w", err)
 		}
 		queryMods = append(queryMods, models.SyntheticDeviceWhere.ID.GT(beforeID))
 	}
 
-	orderBy := "DESC"
+	orderBy := " DESC"
```
Comment (elffjs) [inline internal/repositories/synthetic/synthetic_devices.go:103]: Is this a good change?

**Comment by elffjs** [comment]
@Allyson-English Giving this a nudge.

**Comment by Allyson-English** [comment]
@elffjs thanks for nudging! I made those edits on friday but forgot to push them up because I got distracted by cases of if/ else for the before/ after queries. I removed the use of it here because you pointed out that it's legal but noticed that we're also doing it in these spots and was wondering if we wanted to use this opportunity to make both options allowable in those instances too: [1](https://github.com/DIMO-Network/identity-api/blob/main/internal/repositories/aftermarket/aftermarket_devices.go#L82), [2](https://github.com/DIMO-Network/identity-api/blob/main/internal/repositories/dcn/dcn.go#L127), [3](https://github.com/DIMO-Network/identity-api/blob/main/internal/repositories/reward/rewards.go#L83)

**Comment by elffjs** [comment]
That's a good catch. I can do that in another PR, no worries.

I don't think it's a very _useful_ option but unlike combining first and last it doesn't seem to do any harm.

File: internal/repositories/synthetic/synthetic_devices.go [hunk 4]
```diff
@@ -43,6 +43,7 @@ func ToAPI(sd *models.SyntheticDevice) (*gmodel.SyntheticDevice, error) {
 		IntegrationID: sd.IntegrationID,
 		Address:       common.BytesToAddress(sd.DeviceAddress),
 		MintedAt:      sd.MintedAt,
+		VehicleID:     &sd.VehicleID,
```
Comment (elffjs) [inline internal/repositories/synthetic/synthetic_devices.go:46]: We need to regenerate, I think? This isn't a pointer anymore.

**Review by elffjs — CHANGES_REQUESTED** [review]

File: graph/schema/synthetic.graphqls [hunk 5]
```diff
@@ -64,6 +64,10 @@ type SyntheticDevice implements Node {
   The block timestamp at which this device was minted.
   """
   mintedAt: Time!
+  """
+  The vehicle, if any, with which the synthetic device is paired.
```
Comment (elffjs) [inline graph/schema/synthetic.graphqls:68]: I think we can remove the `if any` here.

File: internal/repositories/synthetic/synthetic_devices.go (hunk 3 above)
Comment (elffjs) [reply internal/repositories/synthetic/synthetic_devices.go:105]: (I do not think it is)

**Pull request merged**